### DISCUSSION TITLE
Lots of "Cannot auto-delete Pod when the associated Volume is not found" messages after 1.5.1 upgrade

### DISCUSSION THREAD

**Discussion by eoprede** [longhorn/longhorn#6419] [General]
Hello,
I have noticed that I am getting lots of the following warning messages after upgrade to 1.5.1:
```time="2023-07-31T01:49:37Z" level=warning msg="Cannot auto-delete Pod when the associated Volume is not found" controller=longhorn-kubernetes-pod error="volume.longhorn.io \"longhorn-vol-pv\" not found" node=slave3 pod=db-74664ff978-87lfz```
I am getting these messages about only 1 volume and this happens to be the only 1 manually created volume I have. Here's how I define it in the k8s yaml:
```---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  labels:
    app: db
  name: db
spec:
  accessModes:
    - ReadWriteOnce
  storageClassName: longhorn
  volumeName: longhorn-vol-pv
  resources:
    requests:
      storage: 2Gi
---
apiVersion: v1
kind: PersistentVolume
metadata:
  name: longhorn-vol-pv
spec:
  capacity:
    storage: 2Gi
  volumeMode: Filesystem
  accessModes:
    - ReadWriteOnce
  storageClassName: longhorn
  csi:
    driver: driver.longhorn.io
    fsType: ext4
    volumeAttributes:
      numberOfReplicas: '3'
      staleReplicaTimeout: '2880'
    volumeHandle: librebackup```

Not sure if there's some kind of issue for 1.5.1 or if this is some kind of misconfiguration on my part. Any suggestions?

**Comment by c3y1huang** [comment]
Longhorn is attempting to get a Volume using the same name as the PV. However, in your case, the volume name "librebackup" is different from the PV name "longhorn-vol-pv."

I've converted this to a [bug](https://github.com/longhorn/longhorn/issues/6420). In the meantime, you can try two possible solutions:
- Disable the setting [Automatically Delete Workload Pod when The Volume Is Detached Unexpectedly](https://longhorn.io/docs/1.5.1/references/settings/#automatically-delete-workload-pod-when-the-volume-is-detached-unexpectedly).
- Recreate the PV with the same name as the volume.

**Reply by eoprede** [reply]
Thank you for the suggestion. Confirming that recreating PV with the same name as the volume solves the issue.
